tests/scalers/neo4j/neo4j_test.go

```diff
@@ -0,0 +1,296 @@
+  containers:
+  - image: tanishabanik/neo4j-demo:0.0.8
+    name: neo4j-demo-activation
+    args: ["neo4j://{{.ScalerName}}.{{.TestNamespace}}.svc.cluster.local:7687",
+	      'CREATE (ac1:Person { name: "Danish", from: "Colombo", popularfor: "singer" }),
+	       (ac2:Person { name: "Saanvi", from: "Delhi", popularfor: "actress" }),
+	       (ac3:Person { name: "Saurav", from: "Mumbai", popularfor: "Badminton" }),
+	       (ac4:Person { name: "Robert", from: "California", popularfor: "Swimming" }),
+	       (ac5:Person { name: "Sezzi", from: "Florida", profession: "Doctor" }),
+	       (ac6:Person { name: "Sally", from: "Texas", profession: "Software Engineer" }),
+	       (ac7:Person { name: "Sohali", from: "Gandhinagar", popularfor: "Tiktok" }),
+	       (ac8:Person { name: "Ranauk", from: "Bangalore", profession: "Entrepreneur" }),
+	       (ac9:Person { name: "Soha", from: "Bhopal", profession: "Charter Accountant" }),
+	       (ac10:Person { name: "Ananya", from: "Paris", popularfor: "Lawn Tennis" }),
+	       (ac11:Person { name: "Anna", from: "Moscow", profession: "Software Engineer" }),
+	       (ac1)-[:FOLLOWS]->(ac2),(ac2)<-[:FOLLOWS]-(ac1),(ac5)-[:FOLLOWS]->(ac1),
+	       (ac6)-[:FOLLOWS]->(ac1),(ac8)-[:FOLLOWS]->(ac1),(ac9)-[:FOLLOWS]->(ac1),
+	       (ac11)-[:FOLLOWS]->(ac1),(ac5)-[:FOLLOWS]->(ac2),(ac6)-[:FOLLOWS]->(ac2),
+	       (ac5)-[:FOLLOWS]->(ac3),(ac6)-[:FOLLOWS]->(ac3),(ac8)-[:FOLLOWS]->(ac3),
+	       (ac9)-[:FOLLOWS]->(ac3),(ac11)-[:FOLLOWS]->(ac3),(ac5)-[:FOLLOWS]->(ac4),
+	       (ac6)-[:FOLLOWS]->(ac4),(ac8)-[:FOLLOWS]->(ac4),(ac9)-[:FOLLOWS]->(ac4),
+	       (ac11)-[:FOLLOWS]->(ac4),(ac5)-[:FOLLOWS]->(ac7),(ac6)-[:FOLLOWS]->(ac7),
+	       (ac5)-[:FOLLOWS]->(ac10),(ac6)-[:FOLLOWS]->(ac10),(ac8)-[:FOLLOWS]->(ac10),
+	       (ac5)<-[:FOLLOWS]-(ac6),(ac5)<-[:FOLLOWS]-(ac8),(ac5)<-[:FOLLOWS]-(ac11),
+	       (ac6)<-[:FOLLOWS]-(ac5),(ac6)<-[:FOLLOWS]-(ac8),(ac6)<-[:FOLLOWS]-(ac11),
+	       (ac6)<-[:FOLLOWS]-(ac9),(ac11)<-[:FOLLOWS]-(ac9),(ac11)<-[:FOLLOWS]-(ac6),
+	       (ac8)<-[:FOLLOWS]-(ac11),(ac8)<-[:FOLLOWS]-(ac9),(ac9)<-[:FOLLOWS]-(ac5),
+	       (ac10)<-[:FOLLOWS]-(ac7),(ac4)<-[:FOLLOWS]-(ac7),(ac3)<-[:FOLLOWS]-(ac7),
+	       (ac3)<-[:FOLLOWS]-(ac10),(ac1)<-[:FOLLOWS]-(ac7),(ac2)<-[:FOLLOWS]-(ac7),
+	       (ac3)<-[:FOLLOWS]-(ac7),(ac3)<-[:FOLLOWS]-(ac1),
+	       (ac10)<-[:FOLLOWS]-(ac3),(ac10)<-[:FOLLOWS]-(ac2),(ac10)<-[:FOLLOWS]-(ac1)
+	       return ac1,ac2,ac3,ac4,ac5,ac6,ac7,ac8,ac9,ac10,ac11',{{.UsernameArg}},{{.PasswordArg}}]
+  restartPolicy: OnFailure
+`
+	KubectlApplyWithTemplate(t, data, "activationPodTemplate", activationPodTemplate)
+}
+
+func testActivation(t *testing.T, kc *kubernetes.Clientset, data templateData) {
+	t.Log("--- testing activation ---")
+	deployPodActivation(t, data)
+	time.Sleep(time.Second * 60)
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, minReplicaCount, 60, 1),
+		"replica count should be %d after 1 minute", minReplicaCount)
```

Comment:
You should test that the workload doesn't scale out in the given time but current code checks that it reach 1 instance in 1 min, which is the opposite. The assertion you have to pass here is this:
```suggestion
	AssertReplicaCountNotChangeDuringTimePeriod(t, kc, deploymentName, testNamespace, minReplicaCount, 60)
```

Reply:
Added the suggested change